File: src/main/java/com/elchologamer/userlogin/listener/OnPlayerJoin.java
package com.elchologamer.userlogin.listener;

import com.elchologamer.userlogin.ULPlayer;
import com.elchologamer.userlogin.UserLogin;
import org.bukkit.event.EventHandler;
import org.bukkit.event.player.PlayerJoinEvent;

public class OnPlayerJoin extends BaseListener {

    private final UserLogin plugin = UserLogin.getPlugin();

    @EventHandler
    public void onPlayerJoin(PlayerJoinEvent e) {
        e.setJoinMessage(null);

        ULPlayer ulPlayer = plugin.getPlayers().getOrCreate(e.getPlayer());
        ulPlayer.onJoin(e);
    }
}

File: src/main/kotlin/com/elchologamer/userlogin/listener/OnPlayerJoin.kt
package com.elchologamer.userlogin.listener

import com.elchologamer.userlogin.UserLogin.Companion.plugin
import org.bukkit.event.EventHandler
import org.bukkit.event.player.PlayerJoinEvent

object OnPlayerJoin : BaseListener() {

    @EventHandler
    fun onPlayerJoin(e: PlayerJoinEvent) {
        e.joinMessage = null
        val ulPlayer = plugin.players.getOrCreate(e.player)
        ulPlayer.onJoin(e)
    }
}